test/utils/wasm3_engine.cpp

```diff
@@ -74,14 +74,18 @@ fizzy::bytes_view Wasm3Engine::get_memory() const
     return {data, size};
 }
 
-void Wasm3Engine::set_memory(fizzy::bytes_view memory)
+bool Wasm3Engine::set_memory(fizzy::bytes_view memory)
 {
-    if (memory.empty())
-        return;
-
     uint32_t size;
     const auto data = m3_GetMemory(m_runtime, &size, 0);
-    std::memcpy(data, memory.data(), size);
+    if (data == nullptr)
+        return memory.empty();
+
+    if (memory.data() == nullptr)
+        std::memset(data, 0, size);
+    else
+        std::memcpy(data, memory.data(), size);
```

Comment:
This is not correct though when `memory` has fewer when `size` bytes.
Will rewrite it a bit.